models/spring-ai-ollama/src/main/java/org/springframework/ai/ollama/api/OllamaApi.java

```diff
@@ -429,8 +468,68 @@ public record ChatRequest(
 			@JsonProperty("stream") Boolean stream,
 			@JsonProperty("format") String format,
 			@JsonProperty("keep_alive") String keepAlive,
-			@JsonProperty("options") Map<String, Object> options) {
+			@JsonProperty("options") Map<String, Object> options,
+			@JsonProperty("tools") List<Tool> tools) {
+
+
+		/**
+		 * Represents a tool the model may call. Currently, only functions are supported as a tool.
+		 *
+		 * @param type The type of the tool. Currently, only 'function' is supported.
+		 * @param function The function definition.
+		 */
+		@JsonInclude(Include.NON_NULL)
+		public record Tool(
+				@JsonProperty("type") Type type,
+				@JsonProperty("function") Function function) {
+
+			/**
+			 * Create a tool of type 'function' and the given function definition.
+			 * @param function function definition.
+			 */
+			@ConstructorBinding
+			public Tool(Function function) {
+				this(Type.FUNCTION, function);
+			}
 
+			/**
+			 * Create a tool of type 'function' and the given function definition.
+			 */
+			public enum Type {
+				/**
+				 * Function tool type.
+				 */
+				@JsonProperty("function") FUNCTION
+			}
+
+			/**
+			 * Function definition.
+			 *
+			 * @param name The name of the function to be called. Must be a-z, A-Z, 0-9, or contain underscores and dashes.
+			 * @param description A description of what the function does, used by the model to choose when and how to call
+			 * the function.
+			 * @param parameters The parameters the functions accepts, described as a JSON Schema object. To describe a
+			 * function that accepts no parameters, provide the value {"type": "object", "properties": {}}.
+			 */
+			public record Function(
```

Comment:
For parity with the Ollama types, this might be named `ToolFunction`

Reply:
It is already nested inside the record `Tool`, so ` new ChatRequest.Tool.ToolFunction` would read awkardly.